beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -185,8 +176,32 @@ pub fn validate_blob_sidecar_for_gossip<T: BeaconChainTypes>(
         });
     }
 
-    // TODO(pawan): should we verify locally that the parent root is correct
-    // or just use whatever the proposer gives us?
+    // We have already verified that the blob is past finalization, so we can
+    // just check fork choice for the block's parent.
+    if let Some(parent_block) = chain
+        .canonical_head
+        .fork_choice_read_lock()
+        .get_block(&block_parent_root)
+    {
+        if parent_block.slot >= blob_slot {
+            return Err(BlobError::BlobIsNotLaterThanParent {
+                blob_slot,
+                parent_slot: parent_block.slot,
+            });
+        }
+    } else {
+        return Err(BlobError::BlobParentUnknown {
```

Comment:
Do we want to send the blobs with the parent unknown error so that we have them in sync for reprocessing? similar to how we handle blocks

Reply:
Yeah I think so. I have a todo for that in `gossip_methods` for this error type.